### PR TITLE
Added server and layer info customization options.

### LINKED ISSUE #170: Allow server info customization.
In [src/info.js](https://github.com/koopjs/FeatureServer/blob/master/src/info.js), the [`serverLayerInfo`](https://github.com/koopjs/FeatureServer/blob/master/src/info.js#L42) function does not allow for the `id`, `defaultVisibility`, `minScale`, or `maxScale` properties to be provided by the metadata object.  I think it would be beneficial to allow for these properties to be overwritten by the metadata object.  Upon further research, the same can be said of the [`renderLayer`](https://github.com/koopjs/FeatureServer/blob/master/src/templates.js#L32) function in [src/templates.js](https://github.com/koopjs/FeatureServer/blob/master/src/templates.js).

A quick and simple fix would be to provide the logic to pull these properties off of the metadata object in these two functions.  However, I wonder if there is an opportunity for consolidation of logic here?  Is there a reason why the [`serverLayerInfo`](https://github.com/koopjs/FeatureServer/blob/master/src/info.js#L42) function shouldn't utilize the [`renderLayer`](https://github.com/koopjs/FeatureServer/blob/master/src/templates.js#L32) function (or even the [`layerInfo`](https://github.com/koopjs/FeatureServer/blob/master/src/info.js#L38) function in [src/info.js](https://github.com/koopjs/FeatureServer/blob/master/src/info.js)) to consistently build up the layer info object?  Another potential improvement could be the inclusion of the `spread` operator (`...`) to allow for all properties to be overwritten dynamically.  This would involve some manipulation of the metadata object, since not all properties currently supported by the metadata object are valid properties on the resulting info object.  However, this could be a good, flexible approach to allow for developers to overwrite any property on the info object using the metadata object.

### PR DESCRIPTION
**Pull request by hparkertt**
Added ability to override id, defaultVisibility, minScale, and maxScale in layer info via metadata object.

*NOTE*: removed hard-coded OBJECTID values in testing in favor of variables, to help account for hash function differences between windows/unix.

Resolves #170

### CODE REVIEW

**Comment by rgwozdz** [comment]
Hey @hparkertt - sorry for the delay, reviewing this ASAP.

File: test/route.js [hunk 1]
```diff
@@ -174,11 +174,15 @@ describe('Routing feature server requests', () => {
   })
 
   describe('Query', () => {
+    // This pattern is used to help insulate bugs that occur from platform-specific logic in farmhash.
+    // The existing hard-coded OBJECTID's were generated by a unix-based farmhash, and would cause
+    // the tests to fail when run on windows.
+    var secondOBJECTID
```
Comment (rgwozdz) [inline test/route.js]: Same comment as above.  Use `beforeEach` to get objectId data you need for the tests and store on `this.currentTest` as shown [here](https://gist.github.com/icirellik/b9968abcecbb9e88dfb2).

**Comment by hparkertt** [comment]
@rgwozdz I'm going to use `before` functions, rather than `beforeEach` functions, just because that feels like the more correct solution.  To me, it doesn't make sense to get the pertinent `OBJECTID`s before every test.  My thought is that we should get all of the `OBJECTID`s prior to testing to ensure they are not changing between tests.  Syntactically, the only difference is that I have to add the `objectIds` property on the `this` object, rather than on the `this.currentTest` object.  I'm going to hold off on pushing another commit until I hear back from you on this.  Here's the before function pattern I'm going to use:

``` javascript
// ./test/query.js
before(function () {
  const response = FeatureServer.query(data, { outFields: 'OBJECTID' })
  this.objectIds = response.features.map(feat => feat.attribtues.OBJECTID)
})

// ./test/route.js
before(function () {
  const response = FeatureServer.query(data, { outFields: 'OBJECTID' })
  this.secondOBJECTID = response.features[1].attributes.OBJECTID
})
```

Also, as an aside, I had tried this pattern before but couldn't get it to work.  After doing a little more research this time around, I discovered that there is a scoping issue when using lambda functions with mocha.  There's a note in their documentation about it [here](https://mochajs.org/#arrow-functions).  It might make sense to create a separate 'housekeeping' ticket (not sure what label you would use) to convert lambda functions over in an attempt to stay consistent.  I can see this as a 'gotcha' that could crop up from time to time if the convention is to use lambda functions in the tests.  Thoughts?

**Comment by rgwozdz** [comment]
@hparkertt - Totally agree on the `before` instead of `beforeEach`, wrote my review comment in haste, good catch.

Yeah, makes sense that it won't work with arrow functions, as the `this` would be from the parent scope.  At this point, we don't have an established convention, other than to use arrow functions unless there is a reason not to.  This seems like a good reason.  Thanks for checking in.

**Comment by rgwozdz** [comment]
Thanks @hparkertt. We'll release this shortly.

**Comment by rgwozdz** [comment]
Apologies for release delay.  Now in 2.22.0.  Thanks again for the contribution.

**Pull request merged**